### DISCUSSION TITLE
Script to Import SuperSlicer/PrusaSlicer Profiles

### DISCUSSION THREAD

**Discussion by theophile** [OrcaSlicer/OrcaSlicer#1636] [General]
I wrote a script to batch convert all my SuperSlicer profiles so OrcaSlicer can use them. It works great for me. I hope it's helpful to others: https://github.com/theophile/SuperSlicer_to_Orca_scripts/

**Comment by MAVProxyUser** [comment]
You sir are a saint!

**Reply by theophile** [reply]
Let me know if you notice any issues.

**Reply by theophile** [reply]
@MAVProxyUser, if you don't mind, please test my standalone script to convert print profiles to OrcaSlicer "process" files:
https://github.com/theophile/SuperSlicer_to_Orca_scripts/blob/print_profiles/superslicer_to_orca-print-process.pl 

That one is intended to work on both PrusaSlicer and SuperSlicer .ini files. I've put it through various tests on my system but since these conversions are significantly more complicated than the filament profiles, I'm sure there are scenarios I haven't accounted for. The biggest challenge is that OrcaSlicer requires either percent values or mm values for some parameters where Super/PrusaSlicer either required the other, or allowed either. And in some cases where Super/PrusaSlicer allowed percents, those slicers used different calculation methods for those percents. I _think_ I caught and handled all those but I might have missed some, especially ones I don't use.

**Comment by MAVProxyUser** [comment]
It's been a decade since I used perl, first thing I noticed was I had to install some modules by hand: 
```cpan install String::Escape ```
```cpan install  Config::Tiny```

Also got this warning, so you may have to hand hold some of you rend users on module dependencies. 
"Warning: You do not have write permission for Perl library directories.

To install modules, you need to configure a local Perl library directory or
escalate your privileges.  CPAN can help you by bootstrapping the local::lib
module or by configuring itself to use 'sudo' (if available).  You may also
resolve this problem manually if you need to customize your setup.

What approach do you want?  (Choose 'local::lib', 'sudo' or 'manual')"

**Reply by theophile** [reply]
Yeah, the README in the GitHub repo explains the need to install the required perl modules. It's got Windows instructions but I'll plan to update to add Linux instructions. I'm not familiar with MacOS but it looks like the command is the same? On MacOS, what happens if you copy and paste this?
`cpan Getopt::Long File::Basename File::Glob File::Spec String::Escape Config::Tiny JSON`

**Reply by MAVProxyUser** [reply]
hah busted... I totally skipped the readme details.

**Comment by MAVProxyUser** [comment]
For a sanity check... you may try this repo. 
https://github.com/DIY141/Inland-Filament-Profiles

Make sure on *nix systems that the wildcard is quoted: 
```
Kevins-MacBook-Air:tmp kfinisterre$ perl superslicer_to_orca-filaments.pl --input "Inland-Filament-Profiles/*.ini" --outdir converted --overwrite  
Translated 'Inland-Filament-Profiles/Inland ABS.ini' to 'convertedInland ABS.json'.
Translated 'Inland-Filament-Profiles/Inland CF-NYLON.ini' to 'convertedInland CF-NYLON.json'.
Translated 'Inland-Filament-Profiles/Inland CF-PLA.ini' to 'convertedInland CF-PLA.json'.
Translated 'Inland-Filament-Profiles/Inland Glass PLA.ini' to 'convertedInland Glass PLA.json'.
Translated 'Inland-Filament-Profiles/Inland Lite PLA.ini' to 'convertedInland Lite PLA.json'.
Translated 'Inland-Filament-Profiles/Inland NYLON.ini' to 'convertedInland NYLON.json'.
Translated 'Inland-Filament-Profiles/Inland PETG+.ini' to 'convertedInland PETG+.json'.
Translated 'Inland-Filament-Profiles/Inland PETG.ini' to 'convertedInland PETG.json'.
Translated 'Inland-Filament-Profiles/Inland PLA+.ini' to 'convertedInland PLA+.json'.
Translated 'Inland-Filament-Profiles/Inland PLA.ini' to 'convertedInland PLA.json'.
Translated 'Inland-Filament-Profiles/Inland PP.ini' to 'convertedInland PP.json'.
Translated 'Inland-Filament-Profiles/Inland Pro-PLA.ini' to 'convertedInland Pro-PLA.json'.
Translated 'Inland-Filament-Profiles/Inland Silk PLA.ini' to 'convertedInland Silk PLA.json'.
Translated 'Inland-Filament-Profiles/Inland Tough PLA.ini' to 'convertedInland Tough PLA.json'.
Translated 'Inland-Filament-Profiles/Inland TPU.ini' to 'convertedInland TPU.json'.
Translated 'Inland-Filament-Profiles/Inland Twinkling PLA.ini' to 'convertedInland Twinkling PLA.json'.
Translated 'Inland-Filament-Profiles/Polymaker Polyflex TPU.ini' to 'convertedPolymaker Polyflex TPU.json'.
Translated 'Inland-Filament-Profiles/Polymaker Polylite ASA.ini' to 'convertedPolymaker Polylite ASA.json'.
Translated 'Inland-Filament-Profiles/Polymaker Polylite PC.ini' to 'convertedPolymaker Polylite PC.json'.
Translated 'Inland-Filament-Profiles/Polymaker Polymax PETG.ini' to 'convertedPolymaker Polymax PETG.json'.
Translated 'Inland-Filament-Profiles/Polymaker Polymax PLA.ini' to 'convertedPolymaker Polymax PLA.json'.
Translated 'Inland-Filament-Profiles/Polymaker Polysmooth PVB.ini' to 'convertedPolymaker Polysmooth PVB.json'.
Translated 'Inland-Filament-Profiles/Polymaker Polywood PLA.ini' to 'convertedPolymaker Polywood PLA.json'.
Translated 'Inland-Filament-Profiles/Proto-Pasta HTPLA.ini' to 'convertedProto-Pasta HTPLA.json'.
```

**Comment by MAVProxyUser** [comment]
The out directory must exist, or it becomes part of the file name.

**Reply by theophile** [reply]
Good catch! I've pushed a commit to fix this.

**Comment by MAVProxyUser** [comment]
I noticed after attempting to mass convert the Inland profiles, that only 3 of them were accepted by Orca. It seemed to delete all the others! If I manually attempt to import them 0 of them are accepted. However if I place them into ~/Library/Application\ Support/OrcaSlicer/user/default/filament/ they kinda work 

```
Kevins-MacBook-Air:tmp kfinisterre$ ls *json 
Inland ABS.json			Inland Lite PLA.json		Inland PLA+.json		Inland Silk PLA.json		Polymaker Polyflex TPU.json	Polymaker Polymax PLA.json
Inland CF-NYLON.json		Inland NYLON.json		Inland PLA.json			Inland TPU.json			Polymaker Polylite ASA.json	Polymaker Polysmooth PVB.json
Inland CF-PLA.json		Inland PETG+.json		Inland PP.json			Inland Tough PLA.json		Polymaker Polylite PC.json	Polymaker Polywood PLA.json
Inland Glass PLA.json		Inland PETG.json		Inland Pro-PLA.json		Inland Twinkling PLA.json	Polymaker Polymax PETG.json	Proto-Pasta HTPLA.json
Kevins-MacBook-Air:tmp kfinisterre$ cp *json  ~/Library/Application\ Support/OrcaSlicer/user/default/filament/
Kevins-MacBook-Air:tmp kfinisterre$ ls  ~/Library/Application\ Support/OrcaSlicer/user/default/filament/
Inland ABS.json			Inland Lite PLA.json		Inland PLA+.json		Inland Silk PLA.json		Polymaker Polyflex TPU.json	Polymaker Polymax PLA.json
Inland CF-NYLON.json		Inland NYLON.json		Inland PLA.json			Inland TPU.json			Polymaker Polylite ASA.json	Polymaker Polysmooth PVB.json
Inland CF-PLA.json		Inland PETG+.json		Inland PP.json			Inland Tough PLA.json		Polymaker Polylite PC.json	Polymaker Polywood PLA.json
Inland Glass PLA.json		Inland PETG.json		Inland Pro-PLA.json		Inland Twinkling PLA.json	Polymaker Polymax PETG.json	Proto-Pasta HTPLA.json
Kevins-MacBook-Air:tmp kfinisterre$ /Applications/Or
OrcaSlicer.app/         Original Prusa Drivers/ 
Kevins-MacBook-Air:tmp kfinisterre$ /Applications/OrcaSlicer.app/Contents/MacOS/OrcaSlicer 
[2023-08-03 13:32:39.806816] [0x00000001ea739e00] [trace]   Initializing StaticPrintConfigs
```
For some reason only the TPU ones remain after the app is launched: 
```
Kevins-MacBook-Air:tmp kfinisterre$ ls  ~/Library/Application\ Support/OrcaSlicer/user/default/filament/
Inland TPU.json			Polymaker Polyflex TPU.json
```

**Reply by theophile** [reply]
That's odd. While I was working on this I did notice that orca slicer would refuse to recognize the converted files if they didn't contain certain required keys, but on my system, it never deleted anything. 

What version of orca slicer are you using? I have been testing this on a recent nightly build, and am putting the version string 1.6.0.0 in the converted files. I wonder if an earlier version of the slicer is choking on these files that claim to be from a later version?

**Comment by MAVProxyUser** [comment]
I'm also using a manual build from a few days ago.

**Reply by theophile** [reply]
It looks like the `start_filament_gcode` parameter is probably the culprit. Now to figure out if it's the script or OrcaSlicer that's choking.

**Reply by MAVProxyUser** [reply]
lets discuss on your repo. https://github.com/theophile/SuperSlicer_to_Orca_scripts/issues/1

**Comment by samiron7567** [comment]
Hey @theophile this looks awesome! Just tried today. I was able to import the filament and print profile. However machine profile is not working. To observe that I opened "/Users/sam/Library/Application Support/OrcaSlicer/user/default/machine" in an explorer. The script exported the file there. Next while I click on OrcaSlicer, it gets cleaned up automatically from the directory.

I tried to import separately next via FILE -> IMPORT -> IMPORT CONFIG and got this. Mine is M1 MacBook Pro
<img width="802" alt="Screenshot 2023-08-22 at 9 55 17 PM" src="https://github.com/SoftFever/OrcaSlicer/assets/45857604/34036e1b-3c9f-4ede-b37b-18d3060df2ed">

**Reply by theophile** [reply]
That happens when the .json has values OrcaSlicer doesn't like. It's hard to anticipate every possible variable so I have to iron these out when they arise. Please check out this issue report: https://github.com/theophile/SuperSlicer_to_Orca_scripts/issues/10 If you could upload the log and an offending .ini file that would help a lot!

**Reply by Mierscheid** [reply]
I have exactly this problem (Orca 1.8.0 to 2.0.0 under Ubuntu and Windows 10 (VM)), only with profiles that the manufacturer or third parties have created **specifically for Orca**.
Could it be that Orca can only import profiles that come as .orca_printer?

I'm sorry if I'm bothering you, but I'm not getting an answer either as a bug report or as a post on Discord. I've tried the craziest things, but I keep getting this useless message.

**Reply by theophile** [reply]
What if you just drop the profiles in their respective folders without importing? I've never found the import function to be useful or necessary.

**Reply by Mierscheid** [reply]
Thank you so much for your help, I really appreciate it!

My own thread is here. I have described everything there.
https://github.com/SoftFever/OrcaSlicer/issues/4944

That is, what happens, if I delete the config-folder, put in the printers and restart 2.0.0:

![Bildschirmfoto vom 2024-04-11 16-34-09](https://github.com/SoftFever/OrcaSlicer/assets/7328047/8c2c6958-54c3-4739-ae07-a517c429c881)

**Comment by samiron7567** [comment]
ok please find the logs.
[debug_Tue_Aug_22_21_39_16_82617.log.0.zip](https://github.com/SoftFever/OrcaSlicer/files/12416271/debug_Tue_Aug_22_21_39_16_82617.log.0.zip)
[Textured-PIF_Standard Voron HF 0.4mm 300mm.ini.zip](https://github.com/SoftFever/OrcaSlicer/files/12416283/Textured-PIF_Standard.Voron.HF.0.4mm.300mm.ini.zip)

**Reply by theophile** [reply]
I've pushed a commit that should fix this issue. Please do a 'git pull' and try the conversion again. Let me know if it still doesn't work after that!

**Reply by samiron7567** [reply]
Yes now it's working. Thank you very much @theophile

**Comment by zetabemol** [comment]
Muchas gracias! Ha sido de mucha ayuda!
